<launch>
    <arg name="pkg" default="$(find unknown_object_recognition)"/>
    <!-- パラメータYAMLファイルを読み込む -->
    <rosparam name="/uor/model_config" command="load" file="$(arg pkg)/config/uor_model.yaml" />
    <rosparam name="/uor/object_class_list" ommand="load" file="$(arg pkg)/config/object_class_list.yaml" />
    <rosparam name="/uor/item_category" command="load" file="$(arg pkg)/config/item_category.yaml" />
    <!-- <rosparam name="/uor/prompt_tmp" command="load" file="$(arg pkg)/config/prompt_temp.yaml" /> -->

    <!-- ノードを起動する -->
    <node name="yolo" pkg="$(arg pkg)" type="yolo.py" output="screen"> </node>
    <node name="uor_yolo_server" pkg="$(arg pkg)" type="uor_yolo_server.py" output="screen"> </node>
    
    <node name="image_server" pkg="$(arg pkg)" type="image_server.py" output="screen"> </node>
    <node name="table_object_server" pkg="$(arg pkg)" type="table_object_server.py" output="screen"> </node>
    
    <node name="uor_gpt_server" pkg="$(arg pkg)" type="uor_gpt_server.py" output="screen"> </node>
    <node name="uor_gemini_server" pkg="$(arg pkg)" type="uor_gemini_server.py" output="screen"> </node>
</launch>
